packages/aws-cdk/lib/api/hotswap/stepfunctions-state-machines.ts

```diff
@@ -60,6 +62,8 @@ class StateMachineHotswapOperation implements HotswapOperation {
 
   public async apply(sdk: ISDK): Promise<any> {
     // not passing the optional properties leaves them unchanged
+    const name = this.stepFunctionResource.stateMachineArn.split(':')[6];
+    print(` ${ICON} hotswapping state machine: %s`, colors.bold(name));
```

Comment:
I don't love the fact that we're putting these `print` statements inside of the hotswap operations themselves.

How about we change the return type of the `apply()` method in `HotswapOperation` to return the identifier of the hotswapped resource (or maybe even resources), and then the `print()` logic will only be inside `hotswap-deployments.ts`?

I feel like that's a nicer separation of concerns than having the `print()` statements here.

Reply:
Moved these back to `hotswap-deployments` as we discussed.